Use core::string_view directly instead of alias

ssam18 opened the pull request:
This PR addresses issue #3046 by replacing all uses of the `string_view` alias with `core::string_view` throughout the codebase for improved clarity in both documentation and source code.

## Changes

- Replaced `string_view` with `core::string_view` in all header files (.hpp)
- Replaced `string_view` with `core::string_view` in all inline implementation files (.ipp)
- Updated documentation files (.qbk) to use `core::string_view`
- Replaced `basic_string_view` with `core::basic_string_view` where applicable
- Retained type aliases in `string_type.hpp` for backward compatibility

## Benefits

- Makes it explicit that Beast uses `boost::core::string_view`
- Improves code clarity and documentation
- Easier for users to understand the library's dependencies

## Files Modified

- 48 header and implementation files
- 4 documentation files
- Total: 52 files with 410 insertions and 347 deletions

Closes #3046

ssam18 committed:
Use core::string_view directly instead of alias

Replaced all uses of string_view and basic_string_view with
core::string_view and core::basic_string_view respectively
throughout the codebase.

Closes #3046

ashtum commented on replace_string_view.py:
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
```
Please exclude `replace_string_view.py` from the committed files.

ssam18 committed:
Fix remaining comment alignment issues

ssam18 committed:
Fix namespace qualification for core::string_view throughout codebase

Use fully qualified boost::core::string_view instead of core::string_view
or beast::string_view to ensure correct namespace resolution in nested
boost::beast namespace contexts. This fixes compilation errors across all
platforms where the compiler was looking for non-existent types like
boost::beast::core::string_view.

Fixes CI compilation failures in PR #3058.

ssam18 committed:
Fix remaining string_view references in RFC7230 files

This fixes macOS compilation errors where the compiler couldn't
resolve the unqualified string_view type in iterator declarations.

ssam18 committed:
Update examples to use boost::core::string_view instead of beast::string_view

ssam18 committed:
Remove boost:: prefix from core::string_view when inside boost namespace and fix comment alignment

ssam18 committed:
Fix MSVC compilation errors in test/doc files by using boost::core::string_view

The test/doc files are not inside the boost namespace, so they need
the fully qualified boost::core::string_view instead of core::string_view.
Changed back to boost::core::string_view in all affected test/doc files.

ssam18 committed:
Fix MSVC compilation errors in example files by using boost::core::string_view

Example files are standalone programs not inside the boost namespace,
so they require the fully qualified boost::core::string_view.
Updated all example files except those in example/doc/ which are
already inside the boost namespace.

ssam18 committed:
Remove unnecessary boost:: namespace prefix in .ipp files

Remove boost:: prefix from core::string_view in implementation files
(.ipp) where code is already inside the boost namespace. This makes
the code cleaner and follows the project's namespace usage conventions.

Also reverted unintended change to release notes as requested by maintainer.

Signed-off-by: Samaresh Kumar Singh <ssam3003@gmail.com>

ssam18 commented on test/doc/core_1_refresher.cpp:
```diff
@@ -33,13 +33,13 @@ snippets()
     //[code_core_1_refresher_1s
         net::const_buffer cb("Hello, world!", 13);
-        assert(string_view(reinterpret_cast<char const*>(
+        assert(boost::core::string_view(reinterpret_cast<char const*>(
```
I tried removing boost namespace qualifier from the test/doc but CI is fails in MSVC. Please take a look at the previous checkins.

ssam18 committed:
removed the boost namespace from test/doc as per code review comment

ssam18 commented:
@ashtum Please let me know if this PR looks good?

ssam18 committed:
Use core::string_view with namespace aliases

Replace boost::core::string_view with core::string_view throughout
example and test/doc directories. Add namespace alias declarations
'namespace core = boost::core;' following project style guidelines.

ssam18 committed:
Fix review comments: align namespace declarations and correct header path comments

ssam18 committed:
Fix: compilation errors, namespace ordering, and comment style

- Add `namespace core = boost::core;` to test/fuzz files that used
  `core::string_view` without it being in scope (compilation failure)
- Revert release_notes.qbk change (reviewers: please don't modify that)
- Fix comment style: `// <boost/core/detail/string_view.hpp>` →
  `// from <boost/beast.hpp>` in all test/doc and example files
- Move `namespace core` before `using tcp` declarations in example
  files where it was incorrectly placed after
- Fix alignment of `namespace core` comment to match surrounding lines

ssam18 committed:
Fix: add namespace core at file scope in core_snippets.cpp

write_string() uses core::string_view at doc_core_snippets namespace
scope, but namespace core was only declared inside fxx(). namespace net
works globally because config.hpp defines it at global scope; core
needs an equivalent declaration. Add it at file scope alongside the
existing using namespace boost::beast.

ssam18 committed:
CI: Add libs/static_assert to Windows get-boost.sh submodule list

libs/regex depends on libs/static_assert via Boost.Build, but it was
missing from the Windows CI submodule checkout, causing all MSVC builds
to fail with 'could not resolve project reference /boost/static_assert'.

ssam18 commented:
@vinniefalco vinniefalco after a long chase, I was able to fix all the failing tests after addressing the review comments. Please take a look at this PR.

ssam18 commented:
@ashtum please review this PR. This has been pending over months.
